restapi/endpoints_test.go

```diff
@@ -0,0 +1,146 @@
+	token     string
+}
+
+func (suite *endpointTestSuite) SetupSuite() {
+	db, err := testutils.MockDatabase()
+	if err != nil {
+		suite.Fail("error while creating database", err)
+	}
+	suite.db = db
+
+	// setup required for jwt authentication
+	viper.SetDefault("auth.jwt_key", "GotFKl1PGgMpLg7D36NiI0hy/gsl6woTCXYdhKATbzc=")
+	viper.SetDefault("auth.jwt_expiry", "2h")
+	viper.SetDefault("auth.jwt_issuer", "endpoint_tests")
+
+	router, err := Router(db)
+	if err != nil {
+		suite.FailNow("error while creating router", err)
+	}
+
+	server := httptest.NewServer(router)
+	suite.endpoint = httpexpect.New(suite.T(), server.URL)
+}
+
+func (suite *endpointTestSuite) SetupTest() {
+	if err := migrations.CreateTables(suite.db); err != nil {
+		suite.Fail("error while creating tables", err)		
+	}
+	if err := migrations.CreateSuperUser(suite.db); err != nil {
+		suite.Fail("error creating super user", err)
+	}
+
+	suite.token = suite.endpoint.POST("/login").
+				WithJSON([]models.User{{
+					Username: "admin",
+					Password: "root",
+				}}).
+	            Expect().
+				Status(http.StatusOK).JSON().Object().Value("payload").String().Raw()
+}
+
+func (suite *endpointTestSuite) TearDownTest() {
+	if err := testutils.DropTables(suite.db); err != nil {
+		suite.Fail("error while dropping tables", err)
+	}
+}
+
+func (suite *endpointTestSuite) TestHealthCheck() {
+	suite.endpoint.GET("/health").
+		Expect().
+		Status(http.StatusOK).JSON().Object().Value("message").String().Equal("Hello World")
+}
+
+func (suite *endpointTestSuite) TestGetAllStories() {
+	suite.endpoint.GET("/stories").
+		Expect().
+		Status(http.StatusOK).
+		Body().Equal( "{\"status\":\"OK\",\"payload\":[]}\n")
+}
+
+func (suite *endpointTestSuite) TestCreateAuthor() {
+	json := []models.Author{
+		{
+			FirstName:     "Edmund",
+			LastName:      "Pevensie",
+			OriginCountry: "Narnia",
+			CurrentCity:   "London",
+		},
+	}
+
+	suite.endpoint.POST("/authors").WithHeader("Authorization", fmt.Sprintf("Bearer %s", suite.token)).
+		WithJSON(json).
+		Expect().
+		Status(http.StatusOK)
+
+	var authorCount int64
+	suite.db.Table("authors").Count(&authorCount)
+	suite.Equal(1, int(authorCount))
+}
+
+func (suite *endpointTestSuite) TestCreateStory(){
+	json := []models.Story{
+		{
+			Title:      "Jane Eyre",
+			Content:    "Classic",
+			AuthorID:    1,
+		},
+	}
+
+	suite.endpoint.POST("/stories").WithHeader("Authorization", "Bearer "+suite.token).
+	WithJSON(json).
+	Expect().
+	Status(http.StatusOK)
+
+	var storyCount int64
+	suite.db.Table("stories").Count(&storyCount)
+	suite.Equal(1, int(storyCount))
+}
+
+func (suite *endpointTestSuite) TestGetStoryByID() {
+	json := []models.Story{
+		{
+			Title:      "Jane Eyre",
+			Content:    "Classic",
+			AuthorID:    1,
+		},
+	}
+
+	suite.endpoint.POST("/stories").WithHeader("Authorization", "Bearer "+suite.token).
+	WithJSON(json).
+	Expect().
+	Status(http.StatusOK)
+	suite.endpoint.GET("/story/1").
+		Expect().
+		Status(http.StatusOK)
```

Comment:
Do we get a payload with some story attributes? We should probably check they're equal to the expected story?

Reply:
yes, adding that in!